samples-generator/lib/createGltfFromPnts.js

```diff
@@ -0,0 +1,177 @@
+         primitives.indices = i;
+     }
+
+     return [{
+         'primitives': [primitives],
+    }];
+ }
+
+/**
+ * Creates accessors from attributeBuffers
+ * @param {Array.<attributeBufferType>} bufferAttributes A list of buffer attributes to convert to GLTF
+ * @param {attributeBufferType=} indexBuffer An optional indexBuffer.
+ * @returns {Object} a buffer views array
+ */
+
+function createAccessorsFromAttributeBuffers(attributeBuffers, indexBuffer) {
+    var componentType;
+    var validComponentType;
+    var normalizedComponentType;
+    var accessors = [];
+    var i = 0;
+
+    for (; i < attributeBuffers.length; ++i) {
+        componentType = attributeBuffers[i].componentType;
+        validComponentType = typeConversion.isValidWebGLDataTypeEnum(componentType);
+        normalizedComponentType = validComponentType ? componentType : typeConversion.componentTypeStringToInteger(componentType);
+        if (typeof attributeBuffers[i].count === 'undefined') {
+          console.assert('oh no');
+        }
+
+        accessors.push({
+            bufferView: i,
+            byteOffset: 0,
+            componentType: normalizedComponentType,
+            type: attributeBuffers[i].type,
+            count: attributeBuffers[i].count,
+            min: attributeBuffers[i].min,
+            max: attributeBuffers[i].max
+        });
+    }
+
+    if (defined(indexBuffer)) {
+        componentType = indexBuffer.componentType;
+        validComponentType = typeConversion.isValidWebGLDataTypeEnum(componentType);
+        normalizedComponentType = validComponentType ? componentType : typeConversion.componentTypeStringToInteger(componentType);
+        accessors.push({
+            bufferView: i,
+            byteOffset: 0,
+            componentType: normalizedComponentType,
+            count: indexBuffer.count,
+            type: indexBuffer.type,
+            min: indexBuffer.min,
+            max: indexBuffer.max
+        });
+    }
+
+    return accessors;
+}
+
+/**
+ * Create a GLTF from PNTS data
+ *
+ * @param {Array.<attributeBufferType>} attributeBuffers An object where each key is the name of a bufferAttribute,
+ *                                                       and each value is another js object with the following keys:
+ * @param {attributeBufferType=} indexBuffer An optional indexBuffer.
+ *
+ * @returns {Object} a GLTF object
+ */
+
+function createGltfFromPnts(attributeBuffers, indexBuffer) {
+    var gltf = {
+        asset : {
+            generator : '3d-tiles-samples-generator',
+            version : '2.0'
+        }
+    };
+
+    gltf.buffers = createAmalgamatedGltfBuffer(attributeBuffers, indexBuffer);
+    gltf.bufferViews = createBufferViewsFromAttributeBuffers(attributeBuffers, indexBuffer);
+    gltf.meshes = createMeshFromAttributeBuffers(attributeBuffers, indexBuffer);
+    gltf.accessors = createAccessorsFromAttributeBuffers(attributeBuffers, indexBuffer);
+    gltf.nodes = [{'mesh': 0}];
```

Comment:
Add the z-up-to-y-up node and RTC node similar to `createGltf`.

(Add the RTC node where applicable, some samples like `PointCloudTimeDynamicWithTransform` and `PointCloudWGS84` set `relativeToCenter` to false)

Little more background: the legacy pnts format is z-up but now that it's moving to gltf it needs to be y-up, hence the z-up to y-up node.